pkg/scheduler/predicates.go

```diff
@@ -255,6 +255,22 @@ func NewServiceAffinityPredicate(podLister PodLister, serviceLister ServiceListe
 	return affinity.CheckServiceAffinity
 }
 
+// TODO:@sdminonne: remove it once checked with @agupta
+func copyValueIfKeyEquals(key string, affinityLabels map[string]string, nodeSelector labels.Selector) bool {
```

Comment:
@brendanburns This function is very much required - I believe that @sdminonne meant that he would remove the comment once it was verified. But I agree, getting the value back and assigning it by the caller seems to make more sense.

@sdminonne In the function you are only considering a few operators. However, I would argue that all the operators (except ExistsOperator) should be considered. Consider a use case where affinity is desired at the "zone" level, but the pod specifies "zone!=z2". Now, in this case, we could still ensure that all service pods get hosted on nodes belonging to the same zone, just not on nodes within the zone "z2". Also, I believe, we do have a way to consider all values in case of InOperator. 

I would modify the affinityLabels variable to be a Selector (or an array of Requirements - whatever works) and any label matches could then be added as a Requirement using the appropriate operator. Also, I would add an additional case for the switch statement (consider this pseudo-code):
```
    case labels.NotEqualsOperator, labels.NotInOperator:
        affinitySelector = append(affinitySelector, r)
        continue
```
The above would ensure that any negative conditions are considered, but at the same (since we don't return and just continue) allow us to look at peer pods for the actual label values.

Let me know if you have any questions.